Code/Framework/AzCore/AzCore/std/string/conversions.h

```diff
@@ -384,6 +459,41 @@ namespace AZStd
         return to_wstring(dest, src.c_str(), src.length());
     }
 
+    template<size_t MaxElementCount1>
+    void to_wstring(AZStd::basic_fixed_string<wstring::value_type, MaxElementCount1, wstring::traits_type>& dest, const char* str, size_t strLen = 0)
+    {
+        dest.clear();
+
+        if (strLen == 0)
+        {
+            strLen = strlen(str);
+        }
+
+        if (strLen > 0)
+        {
+            Internal::WCharTPlatformConverter<>::to_wstring(dest, str, str + strLen);
+        }
+    }
+
+    template<size_t MaxElementCount1, size_t MaxElementCount2>
+    void to_wstring(AZStd::basic_fixed_string<wstring::value_type, MaxElementCount1, wstring::traits_type>& dest, const AZStd::basic_fixed_string<string::value_type, MaxElementCount2, string::traits_type>& src)
+    {
+        return to_wstring(dest, src.c_str(), src.length());
+    }
+
+    inline void to_wstring(wchar_t* dest, size_t destSize, const char* str, size_t srcLen = 0)
+    {
+        if (srcLen == 0)
+        {
+            srcLen = strlen(str) + 1;
+        }
+
+        if (srcLen > 0)
+        {
+            Internal::WCharTPlatformConverter<>::to_wstring(dest, destSize, str, str + srcLen + 1); // copy null terminator
```

Comment:
repeat: The source string should not be assumed to be null-terminated
The null-terminated character should be set after the conversion to a wide char
```suggestion
            Internal::WCharTPlatformConverter<>::to_wstring(dest, destSize, str, str + srcLen);
            dest[srcLen] = '\0';
```

Reply:
same here